orm/src/main/java/org/hibernate/search/query/hibernate/impl/FullTextQueryImpl.java

```diff
@@ -390,12 +390,12 @@ protected boolean isNativeQuery() {
 	}
 
 	@Override
-	public Query setEntity(int position, Object val) {
+	public org.hibernate.query.Query<R> setEntity(int position, Object val) {
```

Comment:
Why would that be simpler?

I didn't put a great deal of thought here, as both alternatives are deprecated, and we don't even implement it ;)